### PR TITLE
Dataloader object key path

### PR DESCRIPTION
**Pull request by trunnelben**
Adding object key path to use for auto generated dataloaders

### CODE REVIEW

File: descriptor/extensions.go [hunk 1]
```diff
@@ -137,6 +137,7 @@ func getLoaderOption(method *Method, value string, many bool) *Loader {
 		Many:              many,
 		RequestFieldPath:  strings.Split(parts[1], "."),
 		ResponseFieldPath: strings.Split(parts[2], "."),
+		ObjectKeyPath:     strings.Split(parts[3], "."),
```
Comment (uforic) [inline descriptor/extensions.go]: When we deploy this, we should make sure that the custard thing won't break the world, we'll do it really quickly in succession, but ideally if there are protos that don't have the 3rd part, it won't fall over

File: descriptor/extensions.go [hunk 2]
```diff
@@ -122,7 +122,7 @@ func getLoaderOption(method *Method, value string, many bool) *Loader {
 	}
 
 	parts := strings.Split(value, ":")
-	if len(parts) != 3 {
+	if len(parts) != 4 {
```
Comment (martinxsliu) [inline descriptor/extensions.go]: We need to update the error message on the following line.

File: descriptor/descriptor.go [hunk 3]
```diff
@@ -111,6 +111,7 @@ type Loader struct {
 	Many              bool
 	RequestFieldPath  []string
 	ResponseFieldPath []string
+	ObjectKeyPath     []string
```
Comment (martinxsliu) [inline descriptor/descriptor.go]: Let's call this `ObjectKeyFieldPath` to be consistent.

File: protobuf/graphql/options.proto [hunk 4]
```diff
@@ -175,6 +175,9 @@ message MethodOptions {
   // same form as the 'load_one' option, except that the key field specified
   // by the 'request_field_path' and the loaded field specified by the
   // 'response_field_path' must both be repeated fields.
+  // Additionally load_many has an extra field: object_key_path
+  // this field tells the dataloader how to map the input to the response object
```
Comment (martinxsliu) [inline protobuf/graphql/options.proto]: I think we can be a bit more descriptive here. Following the example in `load_one`, we can say something like:
```
// Additionally load_many has an extra field: 'object_key_field_path'.
// This is a dot separated field path in the loaded Protobuf message to the
// field representing its primary key. This field can be used by DataLoaders
// to order the response corresponding to the input keys.
```

**Pull request merged**